refactor: restructure backend architecture

HijoDeHonor opened the pull request:
Divided the logic into routes, controllers, services, and repositories as appropriate. I still need to finish organizing the routes for promptGeneration and questions into repositories or utilities, depending on what the QA deems appropriate.

HijoDeHonor committed:
refactor: restructure backend architecture

Divided the logic into routes, controllers, services, and repositories as appropriate.

PiewDev commented on src/routes/questions/questionsService.js:
Let's move the logic in getQuestions to the service

PiewDev commented on src/routes/prompt/promptService.js:
Let's move the logic in generatePrompt to the service

PiewDev commented on src/routes/routine/routineRepository.js:
Move here the logic in ../../prompGeneration/generateRoutine.mjs

PiewDev commented on src/routes/routine/routineRepository.js:
```diff
@@ -0,0 +1,8 @@
+import generateRoutine from "../../prompGeneration/generateRoutine.mjs";
```
```suggestion
import chatSesion from "../../prompGeneration/ia.mjs";
```

PiewDev commented on src/routes/routine/routineRepository.js:
```diff
@@ -0,0 +1,8 @@
+
+export class RoutineRepository {
+  async generateRoutine(prompt) {
+    const routine = await generateRoutine(prompt);
```
```suggestion
    const routine = await chatSession.sendMessage(prompt);
```

PiewDev commented on src/prompGeneration/generateRoutine.mjs:
rename to ia.msj

PiewDev replied:
expose chatSession

PiewDev commented on src/routes/routine/routineService.js:
```diff
@@ -0,0 +1,19 @@
+      throw new Error('Invalid input data');
+    }
+
+    const prompt = await this.promptService.getPrompt(data);
```
check prompt result is valid prompt

HijoDeHonor committed:
refactor: move logic to services and repositories, update import paths, remove unnecessary files

Moved logic into services and repositories as requested by QA. Updated import paths and removed unnecessary files.

PiewDev commented on src/routes/routine/routineRepository.js:
```diff
@@ -0,0 +1,19 @@
+
+export class RoutineRepository {
+  constructor() {
+    this.chatSession = chatSession();
```
```suggestion
    this.initChatSession = chatSession();
```

PiewDev commented on src/routes/routine/routineRepository.js:
```diff
@@ -0,0 +1,19 @@
+
+  async generateRoutine(prompt) {
+    try {
+      const chatSession = await this.chatSession;
```
```suggestion
      const chatSession = await this.initChatSession;
```

HijoDeHonor committed:
Update src/routes/routine/routineRepository.js

change the name for better understanding

Co-authored-by: Pablo Kremser <43791992+PiewDev@users.noreply.github.com>
